app/forms/reactivate_profile_form.rb

```diff
@@ -68,6 +68,12 @@ def valid_password?
     user.valid_password?(password)
   end
 
+  def recovery_code_decrypts?
```

Comment:
I wanted to keep the `rescue` as close to where the exception would be thrown as possible, so it was clear to whomever read it 6 mos from now why there's a `rescue` at all.

The code can be valid and still not be able to decrypt the PII. In fact, that's exactly the use case here. The code supplied is valid as a recovery code, but not as a decryption key.